readme.md

```diff
@@ -691,7 +696,75 @@ coordsToPair(pairToCoords([1, 2])) // [1, 2]
 pairToCoords(coordsToPair({x: 1, y: 2})) // {x: 1, y: 2}
 ```
 
+### Catamorphism
 
+A `reduceRight` function that applies a function against an accumulator and each value of the array (from right-to-left) to reduce it to a single value.
```

Comment:
> each value of the array

What I meant here was that it doesn't need to be an `array` and

> against an accumulator

it doesn't need to be an accumulator

It can be a `Maybe Int`. The catamorphism can help to "break" it down into just `Int`, so that the `Maybe` structure is not preserved.

> reduce it to a single value

It may not be a single value. You can re-implement `map` using `reduce` for list:
```js
const map = reduce((acc, cur) => concat(acc, [cur]))
```

So I think maybe another example not using list will be better here? just to illustrate that Catamorphism is more general than a fold for a list.